### PR TITLE
Add indexing for Epidemic QA dataset

### PR DESCRIPTION
**Pull request by justinborromeo**
The Epidemic QA task (https://bionlp.nlm.nih.gov/epic_qa/) uses a modified version of the CORD-19 dataset.  This PR adds Collection and Generator classes for the dataset.  The dataset comprises an `info.txt` file which contains a list of JSON documents.  The schema of the JSON documents is defined at https://bionlp.nlm.nih.gov/epic_qa/#doc-schema.

The code was validated by:
1. Adding a unit test for the EpidemicQACollection class.
2. Running the indexing task
```
sh target/appassembler/bin/IndexCollection
-collection EpidemicQACollection 
-generator EpidemicQAGenerator
-threads 8
-input collections/epic_qa_cord_2020-06-19_v2
-index indexes/lucene-index-epidemic-qa
-storePositions -storeDocvectors -storeContents -storeRaw -optimize > logs/logs.epidemic-qa.txt
```

3. Using Pyserini's SimpleSearcher to search the generated index (https://gist.github.com/justinborromeo/74a2fb7ca143fedf9ecf9c881f95db16) and checking if the returned documents were relevant to the query.

Update:
I was looking at info.txt and it seems to only contain a subset of the documents.  I sent an email to the mailing list asking why this was given that there's no explanation on the challenge webpage.  Do not merge in the meantime.

Update 2:

I have to update the code to not use info.txt as a list of documents (apparently, it's just a byproduct from preprocessing).  Rather, all json files should be read.
image

### CODE REVIEW

**Comment by ronakice** [comment]
@justinborromeo does this work for the consumer task too?

**Comment by justinborromeo** [comment]
@ronakice I'll test it tonight but it should work since it's the same format.

File: src/main/java/io/anserini/index/generator/EpidemicQAGenerator.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    }
+  }
+
+  public EpidemicQAGenerator(IndexArgs args) {
+    this.args = args;
+  }
+
+  @Override
+  public Document createDocument(EpidemicQACollection.Document covidDoc) throws GeneratorException
+  {
```
Comment (lintool) [inline src/main/java/io/anserini/index/generator/EpidemicQAGenerator.java:64]: braces with preceding line.

File: src/test/java/io/anserini/collection/EpidemicQACollectionTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Before;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+public class EpidemicQACollectionTest  extends DocumentCollectionTest<EpidemicQACollection.Document>
+{
```
Comment (lintool) [inline src/test/java/io/anserini/collection/EpidemicQACollectionTest.java:13]: same

**Review by lintool — CHANGES_REQUESTED** [review]
minor nits. please tweak and I'll merge.

**Comment by justinborromeo** [comment]
Ready for review again.

**Pull request merged**